### PR TITLE
Integration tests of FormatAnalyzer

### PR DESCRIPTION
**Pull request by m09**
This PR includes some data to test against and a sample integration test.

### CODE REVIEW

**Comment by m09** [comment]
@vmarkovtsev PTAL

File: lookout/style/format/tests/test_analyzer_integration.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        self.analyzer = subprocess.Popen(
+            command.split(" "), stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+
+        waiter = Thread(target=self._wait_ready)
+        waiter.start()
+        waiter.join(5)
+        if waiter.is_alive():
+            raise RuntimeError("couldn't start the analyzer correctly")
+        self.analyzer_output = queue.Queue()
+        Thread(target=self._tee).start()
```
Comment (vmarkovtsev) [inline lookout/style/format/tests/test_analyzer_integration.py]: You need to join this thread in tearDown

File: lookout/style/format/tests/test_analyzer_integration.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        subprocess.check_call(command.split(" "))
+        command = ("python3 -m lookout run lookout.style.format --db sqlite:///%s "
+                   "--server localhost:%d --fs %s --log-level DEBUG"
+                   ) % (self.db.name, self.port, self.base_dir)
+        self.analyzer = subprocess.Popen(
+            command.split(" "), stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+
+        waiter = Thread(target=self._wait_ready)
+        waiter.start()
+        waiter.join(5)
```
Comment (vmarkovtsev) [inline lookout/style/format/tests/test_analyzer_integration.py]: We can call `main()` directly without the need for `subprocess`

**Comment by m09** [comment]
I pushed only to rebase and observe logs, no modifs yet

File: lookout/style/format/tests/test_analyzer_integration.py (hunk 2 above)
Comment (m09) [reply lookout/style/format/tests/test_analyzer_integration.py]: Regarding this comment, do you have something in mind specifically? To me subprocess is the easiest to do the 3 required things: wait for initialization, monitor output, send sigint to cleanup.

Comment (vmarkovtsev) [reply lookout/style/format/tests/test_analyzer_integration.py]: Yep, calling the main() function directly. Don't get the question... main() is defined in __main__.py

**Comment by m09** [comment]
Travis fails on the same `lookout` error than #114.

File: lookout/style/format/tests/test_analyzer_integration.py (hunk 2 above)
Comment (vmarkovtsev) [reply lookout/style/format/tests/test_analyzer_integration.py]: I see. It should be possible to find the class instance either through the chain of function locals or the global list of objects. You will be able to shutdown cleanly.

**Comment by vmarkovtsev** [comment]
@m09 I can finish this if you are not working yourself.

**Comment by m09** [comment]
@vmarkovtsev I am not working on this, you can take over if you want! I spotted two things: two print calls somehow ended up in the commit in setupclass and there should be a temp directory for --fs because at this point it pollutes the test dir.

**Comment by vmarkovtsev** [comment]
ok

**Pull request merged**